cmd/fleet/serve.go

```diff
@@ -263,6 +263,13 @@ the way that the Fleet server works.
 				}
 			}
 
+			// If a global enroll secret is provided, replace the current enroll
+			// secret with it. This global value is also used as the default enroll
+			// secret in the middleware if the application is not setup yet.
+			if config.Packaging.GlobalEnrollSecret != "" {
+				ds.ApplyEnrollSecrets(cmd.Context(), nil, []*fleet.EnrollSecret{{Secret: config.Packaging.GlobalEnrollSecret}})
+			}
+
```

Comment:
Is the concern that if you remove the global enroll secret from the config, that it will not revert to the previously stored secret?

Reply:
yes, exactly that!

Reply:
I wonder if instead of applying the secret, we should add a field/method to the datastore like `SetGlobalEnrollSecret` that wouldn't actually persist it to mysql.

Reply:
that is a good idea, so we check if we have an enroll secret set and return that instead of querying the db, right?

Reply:
Yes. You can also return an error from the `Datastore.ApplyEnrollSecrets` if it's been set. But, I'm not sure if that's too much logic in the Datastore.

Reply:
> Is the concern that if you remove the global enroll secret from the config, that it will not revert to the previously stored secret?

Am not following this scenario.

Reply:
Yes, sounds like an edge-case, but worth checking something around it.

How about `if config.Packaging.GlobalEnrollSecret != ""` then load the enroll secret and if it exists and is different maybe error out. The user would need to remove the secret from the DB manually, or... we add a separate config `FLEET_PACKAGING_GLOBAL_ENROLL_SECRET_ALLOW_OVERRIDE` to allow overriding it.

Reply:
I like that, and I'd favor erroring out to keep the config file as simple as possible.